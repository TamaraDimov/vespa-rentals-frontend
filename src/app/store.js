import { configureStore } from '@reduxjs/toolkit';
import userSlice from '../redux/reducers/userSlice';
import motorcycleReducer from '../redux/reducers/motorcycleSlice';
import reservationReducer from '../redux/reducers/reservationSlice';

const store = configureStore({
  reducer: {
    motorcycle: motorcycleReducer,
    reservation: reservationReducer,
    user: userSlice,
  },
});

export default store;
