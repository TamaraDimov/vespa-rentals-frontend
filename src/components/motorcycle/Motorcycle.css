@import url('https://fonts.cdnfonts.com/css/dyson-modern');

.moto-main-container {
  background: url(../../assets/home-bg.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  color: white;
  height: 100vh;
  display: flex;
  padding: 0 1rem;
}

.moto-inner-container {
  margin: 4% auto;
  background-color: rgba(0, 0, 0, 0.64);
  border-radius: 50px;
  width: 75vw;
  padding: 2% 4%;
  border: 2px solid white;
  text-align: center;
}

.latest-model-title {
  font-size: 3vw;
  font-weight: bold;
  margin-top: 20px;
  margin-bottom: 20px;
  text-align: center;
}

.latest-models-list {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  max-height: 60vh;
  overflow-y: auto;
}

.motorcycle-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  padding: 20px;
  border: 1px solid black;
  border-radius: 10px;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
}

.motorcycle-item:hover {
  transform: scale(1.1);
}

.single-motor-list {
  width: 300px;
  margin-bottom: 2rem;
  background-color: #fff;
  border-radius: 2rem;
  color: black;
  border: 4px solid #e35e0a;
  box-shadow: 0 0 5px #ddb49b, 0 0 10px #ec8849, 0 0 20px #e35e0a, 0 0 40px;
  animation: glow 1s ease-in-out infinite alternate;
}

@keyframes glow {
  from {
    box-shadow:
      0 0 2px #d89b74,
      0 0 8px #f47e35,
      0 0 10px #e35e0a,
      0 0 20px #e35e0a;
  }

  to {
    box-shadow:
      0 0 5px #e35e0a,
      0 0 10px #e35e0a,
      0 0 15px #e35e0a,
      0 0 20px #e35e0a;
  }
}

.single-motor-image {
  width: 400px;
  object-fit: scale-down;
}

.single-motor-name {
  font-size: 24px;
  font-weight: bold;
  padding: 4%;
  background-color: black;
  border-top-left-radius: 1rem;
  border-top-right-radius: 1rem;
  color: white;
  margin-bottom: 0;
}

.single-motor-model {
  font-size: 20px;
  font-family: 'Dyson Sans Modern', sans-serif;
  color: #f7f7f7;
  font-weight: bold;
  padding: 4%;
  background-color: black;
}

.single-motor-description {
  font-size: 1vw;
  font-weight: 600;
  margin: 2%;
}

.single-motor-socials {
  width: 60%;
  display: flex;
  justify-content: space-around;
  margin: 20px auto;
}

.single-motor-socials img {
  width: 30px;
  height: 30px;
  cursor: pointer;
}

@media (max-width: 768px) {
  .moto-inner-container {
    width: 70vw;
  }

  .latest-model-title {
    font-size: 5vw;
  }

  .latest-models-list {
    max-height: 80vh;
  }

  .motorcycle-item {
    width: 100%;
  }

  .single-motor-list {
    width: 90%;
  }

  .single-motor-image {
    width: 100%;
  }

  .single-motor-name {
    font-size: 20px;
  }

  .single-motor-model {
    font-size: 16px;
  }

  .single-motor-description {
    font-size: 14px;
  }

  .single-motor-socials {
    width: 80%;
  }
}

@media (min-width: 1032px) {
  .latest-model-title {
    font-size: 3vw;
    font-weight: bold;
    margin-top: 2px;
    margin-bottom: 15px;
    text-align: center;
  }

  .latest-models-list {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    overflow-y: auto;
  }
}

@media (max-width: 767px) and (min-width: 400px) {
  /* CSS rules for screen widths between 400px and 767px */

  /* Add your CSS styles here */
  .single-motor-list {
    width: 75%;
    margin-bottom: 4vw;
    max-width: 300px;
  }
}

@media (min-width: 1320px) {
  .latest-model-title {
    font-size: 3vw;
    font-weight: bold;
    margin-top: 2px;
    margin-bottom: 15px;
    text-align: center;
  }

  .latest-models-list {
    display: flex;
    justify-content: space-around;
    flex-wrap: wrap;
    overflow-y: auto;
    height: 670px;
  }
}
