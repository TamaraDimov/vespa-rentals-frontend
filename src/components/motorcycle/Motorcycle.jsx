import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';
import { fetchMotorcycle, fetchUser } from '../../redux/reducers/motorcycleSlice';
import Item from './subcomponents/MotorcycleItem';
import './Motorcycle.css';

export default function Motorcycle() {
  const dispatch = useDispatch();
  const motorcycle = useSelector((state) => state.motorcycle);
  const { user } = useSelector((state) => state.user);
  const token = user && user.user && user.user.token;
  const navigate = useNavigate();

  useEffect(() => {
    if (token) {
      dispatch(fetchUser(token));
      dispatch(fetchMotorcycle(token));
    } else {
      navigate('/');
    }
  }, [token, dispatch, navigate]);
  if (motorcycle.status === 200) {
    const Succes = () => motorcycle.motorcycle.map((item) => (
      <Item
        key={item.id}
        id={item.id.toString()}
        name={item.name}
        photo={item.photo}
        model={item.model}
        description={item.description}
      />
    ));

    return (
      <div className="moto-main-container">
        <div className="moto-inner-container">
          <h1 className="latest-model-title">LATEST MODELS</h1>
          <div className="latest-models-list">
            {motorcycle.isLoading ? <Succes /> : 'Loading...'}
          </div>
        </div>
      </div>
    );
  }
}
