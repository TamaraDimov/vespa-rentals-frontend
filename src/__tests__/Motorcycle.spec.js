import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import Motorcycle from '../components/motorcycle/Motorcycle';

describe('Motorcycle component', () => {
  test('renders correctly', () => {
    const component = create(
      <Provider store={store}>
        <Router>
          <Motorcycle />
        </Router>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
