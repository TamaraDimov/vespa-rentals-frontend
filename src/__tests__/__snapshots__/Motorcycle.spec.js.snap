// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Motorcycle component renders correctly 1`] = `null`;
